nifi/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteOnAttribute.java

```diff
@@ -103,6 +105,8 @@
     private volatile String configuredRouteStrategy = ROUTE_STRATEGY.getDefaultValue();
     private volatile Set<String> dynamicPropertyNames = new HashSet<>();
 
+    private final Map<Relationship, PropertyValue> propertyMap = new HashMap<>();
```

Comment:
Map is both modified and read by multiple threads. This is a thread-safety issue. If we change the 'final' modifier to 'volatile' it should address this (assuming that the @OnUnscheduled annotation is changed to @OnStopped), since it is modified only in the @OnScheduled and @OnStopped method, and no other threads can be in this class at that time.